File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    dic = {player: i for i, player in enumerate(players)}
    for call in callings:
        tmp = dic[call]
        dic[call] -= 1
        dic[players[tmp-1]] += 1
        players[tmp-1], players[tmp]=players[tmp],players[tmp-1]
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.swift
import Foundation

func solution(_ players:[String], _ callings:[String]) -> [String] {
    var player = players
    var rank: [String: Int] = [:]
    
    for (i, p) in player.enumerated() {
        rank[p] = i
    }
    
    for calling in callings {
        let cur = rank[calling]!
        let prevName = player[cur-1]
        
        player[cur-1] = calling
        player[cur] = prevName
        
        rank[calling] = cur - 1
        rank[prevName] = cur
    }
    
    return player
}